scripts/benchmark_multithreaded.py

```diff
@@ -0,0 +1,205 @@
+# To prevent _tkinter.TclError: https://stackoverflow.com/a/37605654
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+
+import os
+import argparse
+import multiprocessing
+import subprocess
+import json
+
+from datetime import datetime
+
+
+# This script executes the Hyrise TPCH benchmarks multiple times over a specified range of cores,
+# and produces a plot of the results. You'll have to provide the path to the hyriseBenchmarkTPCH executable
+# as argument, as well as all parameters you want to be passed to the executable, like --scale, --chunk_size, etc.
+# You'll also have to specify the number of query runs as either a fixed number for all cores (--fixed-runs),
+# or a number of runs per core (--runs-per-core).
+# You can also specify the cores to be benchmarked, otherwise a default range will be used.
+#
+# Example usage 1:
+# python3 ./scripts/benchmark_multithreaded.py ./build-release/hyriseBenchmarkTPCH --runs 1000 -v
+#
+# Example usage 2:
+# python3 ./scripts/benchmark_multithreaded.py ./build-release/hyriseBenchmarkTPCH --runs 1000 -v --scale 1 --cores 10 20 30 --queries 1 3 6 12 --clients 20
+
+
+MAX_CORE_COUNT = multiprocessing.cpu_count()
+DEFAULT_TPCH_QUERIES = [query for query in range(1, 23) if query != 15] # Exclude query 15 which is not supported in our multithreaded benchmarks
+
+def get_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v', '--verbose', action='store_true', help='Print log messages')
+    parser.add_argument('-q', '--queries', action='store', type=int, metavar='Q', nargs='+', help='Specify the TPC-H queries that will be benchmarked')
+    parser.add_argument('--result-dir', action='store', type=str, metavar='DIR', default='results', help='Directory where the result folder will be stored (default: \'results/\')')
+    parser.add_argument('--result-name', action='store', type=str, metavar='NAME', help='Directory where the actual results will be stored (default: current datetime)')
+    parser.add_argument('executable', action='store', type=str, metavar='EXECUTABLE', help='hyriseBenchmarkTPCH executable')
+    parser.add_argument('--clients', action='store', type=int, metavar='C', default=10, help='The number of clients that schedule queries in parallel')
```

Comment:
This doesn't show up in the help

Reply:
The help for `--clients`? For me it does:

```
lanice:opossumdb (leander/1005_multithreaded_benchmarks) $ ./scripts/benchmark_multithreaded.py --help
usage: benchmark_multithreaded.py [-h] [-v] [-q Q [Q ...]] [--result-dir DIR]
                                  [--result-name NAME] [--clients C]
                                  [-c C [C ...] | --max-cores C]
                                  (-r N | --runs-per-core N)
                                  EXECUTABLE

positional arguments:
  EXECUTABLE            hyriseBenchmarkTPCH executable

optional arguments:
  -h, --help            show this help message and exit
  -v, --verbose         Print log messages
  -q Q [Q ...], --queries Q [Q ...]
                        Specify the TPC-H queries that will be benchmarked
  --result-dir DIR      Directory where the result folder will be stored
                        (default: 'results/')
  --result-name NAME    Directory where the actual results will be stored
                        (default: current datetime)
  --clients C           The number of clients that schedule queries in
                        parallel
  -c C [C ...], --cores C [C ...]
                        List of cores to be used for the benchmarks
  --max-cores C         Number of cores up to which the benchmarks will be
                        executed
  -r N, --runs N        Fixed number of (max) runs each query is executed
  --runs-per-core N     Number of (max) runs per core each query is executed
```

Reply:
Maybe I am blind. Or I was on the wrong branch. Or... some other lame excuse.